asynctest/case.py

```diff
@@ -340,6 +340,26 @@ class FunctionTestCase(TestCase, unittest.FunctionTestCase):
     """
 
 
+class ClockedTestCase(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.loop.time = lambda: self.time
+        self.time = 0
+
+    def advance(self, seconds):
+        assert seconds >= 0
+        self._drain_loop()
+        self.time += seconds
+        self._drain_loop()
+
+    def _drain_loop(self):
+        while self.loop._ready or any(map(
+                lambda handle: handle._when <= self.time,
+                self.loop._scheduled)):
+            self.loop._run_once()
```

Comment:
The loop is not re-entrant, so ``advance()`` can't be called on top of the loop (from a coroutine test, for example), transforming ``_drain_loop()`` and ``advance()`` into coroutine functions and ``await  asyncio.sleep(0)`` until there are no callbacks left to run is probably a better solution (like with ``asyncio.helpers.exhaust_loop``).

On the other hand, would it be a bad idea to run ``advance()`` asynchronously during the test?